examples/with-segment-analytics-pages-router/components/analytics.tsx

```diff
@@ -0,0 +1,13 @@
+import { useEffect } from 'react'
+import { useRouter } from 'next/router'
+import { analytics } from '@/lib/segment'
+
+export default function Analytics() {
+  const { asPath } = useRouter()
+
+  useEffect(() => {
+    analytics.page()
+  }, [asPath])
```

Comment:
https://nextjs.org/docs/pages/api-reference/functions/use-router#routerevents

This might be better to do because `asPath` might represent the actual file path instead of a changed route. If you want to keep using `asPath`, I'd conditionally call `page()` only if `router.isReady` is true.

```suggestion
  const router = useRouter()

  useEffect(() => {
    router.events.on('routeChangeComplete', () => analytics.page());
    return () => {
      router.events.off('routeChangeComplete', () => analytics.page());
    }
  }, [asPath])
```

Reply:
updated example. Alternatively, you can stick with `asPath`, I'd just do so conditionally after `router.isReady` is true.

Reply:
I got it working with `routeChangeComplete` thanks @kylemh